os/kernel/environ/env_setenv.c

```diff
@@ -195,7 +195,7 @@ int setenv(FAR const char *name, FAR const char *value, int overwrite)
 
 	/* Now, put the new name=value string into the environment buffer */
 
-	sprintf(pvar, "%s=%s", name, value);
+	snprintf(pvar, sizeof(pvar), "%s=%s", name, value);
```

Comment:
If env is already existed, sizeof(pvar) is wrong. How about change it to varlen?